projector-plugin/src/main/kotlin/SessionDialog.kt

```diff
@@ -0,0 +1,245 @@
+  private val myHostsList: HostsList = HostsList(ProjectorService.host)
+  private val portEditor: PortEditor = PortEditor(ProjectorService.port)
+  private val rwTokenEditor: TokenEditor = TokenEditor("read-write")
+  private val roTokenEditor: TokenEditor = TokenEditor("read-only")
+  private val rwInvitationLink: InivitationLink = InivitationLink()
+  private val roInvitationLink: InivitationLink = InivitationLink()
+  private val roInvitationTitle = JLabel("Read Only Link:")
+
+  val rwToken: String? get() = rwTokenEditor.token
+  val roToken: String? get() = roTokenEditor.token
+  val host: String get() = myHostsList.selected
+  val port: String get() = portEditor.value
+
+  init {
+    if (ProjectorService.isSessionRunning) {
+      title = "Edit Current Session Parameters"
+      description.text = "<html>The current session has already started.<br>Do you want to change passwords?"
+      myOKAction.putValue(Action.NAME, "Save")
+
+      myHostsList.isEnabled = false
+      portEditor.isEnabled = false
+
+      rwTokenEditor.token = ProjectorService.currentSession.rwToken
+      roTokenEditor.token = ProjectorService.currentSession.roToken
+    } else {
+      title = "Start Remote Access to IDE"
+      description.text = "<html>Config remote access to IDE.<br>Please check your connection parameters:"
+      myOKAction.putValue(Action.NAME, "Start")
+
+      rwTokenEditor.token = generatePassword()
+      roTokenEditor.token = generatePassword()
+    }
+
+    myHostsList.onChange = ::updateInvitationLinks
+    portEditor.onChange = ::updateInvitationLinks
+    rwTokenEditor.onChange = ::updateInvitationLinks
+    roTokenEditor.onChange = ::updateInvitationLinks
+
+    updateInvitationLinks()
+    setResizable(false)
+    init()
+  }
+
+  override fun createCenterPanel(): JComponent? {
+    val addressPanel = JPanel()
+    LinearPanelBuilder(addressPanel)
+      .addNextComponent(myHostsList, width = 0.5, rightGap = 35)
+      .addNextComponent(portEditor, width = 0.5)
+
+    val panel = JPanel()
+    LinearPanelBuilder(panel)
+      .addNextComponent(description, gridCount = 2, bottomGap = 5)
+      .startNextLine().addNextComponent(addressPanel, 2)
+      .startNextLine().addNextComponent(rwTokenEditor, 2)
+      .startNextLine().addNextComponent(roTokenEditor, 2)
+      .startNextLine().addNextComponent(JLabel("Invitation Links:"), gridCount = 2, topGap = 5, bottomGap = 5)
+      .startNextLine().addNextComponent(JLabel("Full Access Link:")).addNextComponent(rwInvitationLink)
+      .startNextLine().addNextComponent(roInvitationTitle).addNextComponent(roInvitationLink)
+    return panel
+  }
+
+  private fun updateInvitationLinks() {
+    rwInvitationLink.update(host, port, rwTokenEditor.token)
+    roInvitationLink.update(host, port, roTokenEditor.token)
+
+    if (rwTokenEditor.token == roTokenEditor.token) {
+      // Hide RO invitation section when tokens are equal - user will always get RW access in this case.
```

Comment:
Great! I think also we should display a small label-hint that says something like "to disable read-only access, set its settings to the ones of read-write access". This label or its part can be clickable (displayed like hyperlink), and the click will cause automatic copy.